addons/pos_self_order/models/pos_order.py

```diff
@@ -25,6 +24,20 @@ def _export_for_ui(self, orderline):
 class PosOrder(models.Model):
     _inherit = "pos.order"
 
+    @api.model
+    def create_from_ui(self, orders, draft=False):
+        orders = super().create_from_ui(orders, draft)
+        order_ids = self.env['pos.order'].browse([order['id'] for order in orders])
+
+        for order in order_ids:
+            if order.state != 'draft' and order.state != 'cancel' and order.access_token:
+                self.env['bus.bus']._sendone(f'self_order-{order.access_token}', 'ORDER_STATE_CHANGED', {
```

Comment:
When the order is deleted by the cashier, the client (self-orderer) can continue adding products but when he submits the order, he'll be told that he's unauthorized. Note that a workflow that the client wants to make corrections at the beginning can be common. So perhaps we should also take into account the "deletion" of order from the cashier?

Reply:
I agree with that this will be done after rebasing on this PR: https://github.com/odoo/odoo/pull/125743